ocaml/xapi/xapi_vm_migrate.ml

```diff
@@ -892,6 +892,8 @@ let migrate_send'  ~__context ~vm ~dest ~live ~vdi_map ~vif_map ~options =
           in
           let vm = List.hd vms in
           let () = if ha_always_run_reset then XenAPI.VM.set_ha_always_run ~rpc:remote.rpc ~session_id:remote.session ~self:vm ~value:false in
+          (* Reserve resources for the new VM on the destination pool's host *)
+          let () = XenAPI.Host.allocate_resources_for_vm remote.rpc remote.session remote.dest_host vm true in
```

Comment:
The call to `allocate_vm_to_host` requires a cleanup operation in case of failure. Looking at message forwarding this is `finally_clear_host_operation`, that is usually called in pair with `clear_scheduled_do_be_resident_on`. We need to add a sister API call to forward the cleanup in case of failures.